src/plugin.cpp

```diff
@@ -2,124 +2,295 @@
 #include "illixr/error_util.hpp"
 
 #include <algorithm>
+#include <boost/algorithm/string/join.hpp>
 #include <cstdlib>
 #include <iostream>
+#include <pwd.h>
 #include <sstream>
+#include <stdexcept>
 #include <unistd.h>
 #include <yaml-cpp/yaml.h>
 
+namespace ILLIXR {
+struct Dependency {
+    std::string                                     name;
+    std::map<std::string, std::vector<std::string>> deps;
+
+    bool operator==(const Dependency& rhs) const {
+        return name == rhs.name;
+    }
+
+    bool operator==(const std::string& rhs) const {
+        return name == rhs;
+    }
+};
+} // namespace ILLIXR
+
+namespace YAML {
+template<>
+struct convert<ILLIXR::Dependency> {
```

Comment:
Just on naming conventions, I think we're doing lower case from struct names?

Reply:
ok, will fix on merge.